modules/hdbt_content/hdbt_content.module

```diff
@@ -75,3 +75,39 @@ function hdbt_content_language_switch_links_alter(array &$links) {
     }
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function hdbt_content_form_views_exposed_form_alter(&$form, $form_state) {
+
+  // Handle only Unit search view form at this point.
+  if ($form['#id'] !== 'views-exposed-form-unit-search-block') {
+    return;
+  }
+
+  // Get view from form state.
+  $view = $form_state->getStorage()['view'];
+  $current_language = \Drupal::languageManager()->getCurrentLanguage()->getId();
```

Comment:
Current language should use content language: `\Drupal::languageManager()->getCurrentLanguage(LanguageInterface::TYPE_CONTENT)->getId();`